trinity/protocol/eth/requests.py

```diff
@@ -30,13 +32,33 @@ def __init__(self,
         self.skip = skip
         self.reverse = reverse
 
-    def validate_response(self, response: Any) -> None:
+
+class NodeDataRequest(BaseRequest):
+    node_keys_cache: Tuple[Hash32, ...]
+
+    def __init__(self, node_hashes: Tuple[Hash32, ...]) -> None:
+        self.node_hashes = node_hashes
+
+    def validate_response(self, response: Tuple[Tuple[Hash32, bytes], ...]) -> None:
         """
         Core `Request` API used for validation.
         """
-        if not isinstance(response, tuple):
-            raise ValidationError("Response to `HeaderRequest` must be a tuple")
-        elif not all(isinstance(item, BlockHeader) for item in response):
-            raise ValidationError("Response must be a tuple of `BlockHeader` objects")
+        return self.validate_node_data(response)
+
+    def validate_node_data(self, node_data: Tuple[Tuple[Hash32, bytes], ...]) -> None:
```

Comment:
Not really a problem, but why the extra layer of function call here? Seems reasonable to inline this in `validate_response`

Reply:
I think is a left-over from some type checking hoop jumping which has now been removed/resolved.  Will remove the extra function call layer.